plonk/src/circuit/basic.rs

```diff
@@ -1171,24 +1195,45 @@ where
     }
 
     fn compute_key_table_polynomial(&self) -> Result<DensePolynomial<F>, PlonkError> {
-        let key_table = self.compute_key_table()?;
+        self.check_plonk_type(PlonkType::UltraPlonk)?;
+        self.check_finalize_flag(true)?;
+        let domain = &self.eval_domain;
+        Ok(DensePolynomial::from_coefficients_vec(
+            domain.ifft(&self.table_key_vec()),
+        ))
+    }
+
+    fn compute_table_dom_sep_polynomial(&self) -> Result<DensePolynomial<F>, PlonkError> {
+        self.check_plonk_type(PlonkType::UltraPlonk)?;
+        self.check_finalize_flag(true)?;
+        let domain = &self.eval_domain;
+        Ok(DensePolynomial::from_coefficients_vec(
+            domain.ifft(&self.table_dom_sep_vec()),
```

Comment:
Not a comment but just to aid my own understanding. 
So here we are doing two more `ifft`s. 
It seems that the degree `n` is much larger than `dom_sep.len()`.
Is it still better to do iffts (which costs the same regardless `dom_sep.len()`)? Can we KZG-commit a smaller poly here?